dlc/src/digit_decomposition.rs

```diff
@@ -0,0 +1,673 @@
+//! Utility functions to decompose numeric outcome values
+
+use super::{Error, Payout, RangeOutcome};
+
+struct PrefixRange {
+    start: Vec<usize>,
+    end: Vec<usize>,
+    prefix: Vec<usize>,
+}
+
+/// Decompose a numeric value into digits in the specified base. If the decomposed
+/// value contains less than `nb_digits`, zeroes will be prepended to reach `nb_digits`
+/// size.
+pub fn decompose_value(mut value: usize, base: usize, nb_digits: usize) -> Vec<usize> {
```

Comment:
I'm new to Rust so genuine interest question: Is it good practice to have `mut value` here in place of making a mutable copy?

Reply:
Because `usize` is a primitive type it is actually copied when passed by value (if that makes sense). The `mut` here just indicates that the variable `value` can be mutated within the function. I don't know if it's good practice or not, but since it's copied I don't see what could be bad about it.